regcomp.c

```diff
@@ -25077,7 +25078,9 @@ S_handle_names_wildcard(pTHX_ const char * wname, /* wildcard name to match */
 
     /* Compile the subpattern consisting of the name being looked for */
     subpattern_re = compile_wildcard(wname, wname_len, FALSE /* /-i */ );
-    must = re_intuit_string(subpattern_re);
+
+    /* ref count incremented because regexec.c decrements it. GH #17734 */
+    must = SvREFCNT_inc(re_intuit_string(subpattern_re));
```

Comment:
If anything between here and line 25417 can catchably die, this will leak. Given there's a CALLREGEXEC in there, I think you have to assume that's possible.

Reply:
You're right.  I changed this to make and use a mortalized copy instead